### ISSUE TITLE
Idea: thrower hulk

### ISSUE THREAD

**Issue by aeoo** [CleverRaven/Cataclysm-DDA#23849]
I had an idea for a new hulk that throws nearby zombies at the player.  This new hulk would be relatively common, and its primary offensive technique would be to throw zombies across the entire reality bubble at the player.  If the throw was an accurate one, both the zombie and the player take damage.  Otherwise the zombie lands nearby and provides a near immediate melee threat to the player.  Since these hulks specialize in throwing, they can also throw two zombies at once, one in each arm.  However, each zombie gets a chance to hit the player, but only one hit should be possible.  Whether or not one of the thrown zombies hit the player they land nearby to immediately continue to attack the player.

This hulk should probably take it a bit of time to get ready for a throw, and more time if it decides to throw two at once.

When under attack, this hulk likes to defensively throw the player back to avoid melee.

Also if there aren't any zombies nearby to throw, it may resort to throwing the various junk that's nearby, including small boulders, which may be a bit unlucky if it happens.  These hulks should have a strong preference toward throwing other zombies at the player and only look for nearby junk to throw if they're out of options.

If there is nothing nearby to throw at the player, it should retreat away from the player.

**Comment by Theundyingcode** [comment]
>And sniper-throwing zombies is just silly.

I second this.  They should use the same throwing mechanics as everyone else.  The only difference is they should use some ridiculous value for the strength parameter.

> Also if there aren't any zombies nearby to throw, it may resort to throwing the various junk that's nearby,

I think this is a good idea however, they are hulks; I should never see "The Hulk throws panties at you" or  even "The Hulk throws a laptop at you."  They should ignore anything smaller than a certain size that doesn't at least look heavy.  Rather than throwing random junk, they should rip off a car door and then throw the rest of the car at you.

> have to underline the "common-type", as throwing certain types of zombies that benefit from ability to attack the player from the distance or perform other support roles would be quite bad. 

I thought that the point of Hulks was that they were HULKS.  Let alone having the intelligence to distinguish the zombie types, they shouldn't even have the F***s to care about what abilities they squishy object in front of them might have. 

> I'd argue we do not need special new enemy type for the whole thing, but it'd be a great way to further flesh out currently available hulks.

I'd make it a hulk evolution.  Their is definitely room for endgame difficulty increases; I scraped together a  'nopenemies' mod full of hulk evolutions it would fit in great with.

**Comment by FulcrumA** [comment]
>I thought that the point of Hulks was that they were HULKS. Let alone having the intelligence to distinguish the zombie types, they shouldn't even have the F***s to care about what abilities they squishy object in front of them might have.

True. I think of this limitation purely from the standpoint of gameplay balance. It may be funny, but it may also make things easier if said hulk, for example, grabs and throws a necromancer ahead of the rest of the horde, thus denying them a very powerful support.

>Their is definitely room for endgame difficulty increases; I scraped together a 'nopenemies' mod full of hulk evolutions it would fit in great with.
Perhaps, but it could also be an underserved nightmare for people who didn't get to the late game given how evolution works. Hulks are already a big threat for insufficiently equipped character, but now with enough of bad luck one may meet upgraded hulks as well. That's why rather than having hulks have a chance to turn into this particular one, I'd give them the ability while keeping their stats otherwise as they are - which may make them slightly more powerful, but not too much, while making them also manageable.

For the sake of balance, I'd keep really powerful enemies a thing limited to special locations player takes up as a challenge themselves. Things like mine's the Thing or Amigara horrors.

**Comment by Theundyingcode** [comment]
> I'd keep really powerful enemies a thing limited to special locations player takes up as a challenge themselves. 

I keep it as a mod that I add into the world save data sometime after I take a stroll through a metropolis during the middle of the day cuz I'm too lazy to go back to my base and grab something I forgot.

**Comment by kevingranade** [comment]
> True. I think of this limitation purely from the standpoint of gameplay balance.

You're overthinking it, zombies are supposed to be strong but dumb, that's the whole point.  If the player can trick the thrower into hurting other zombies, that's perfect.

> For the sake of balance, I'd keep really powerful enemies a thing limited to special locations player takes up as a challenge themselves.

My focus is on preventing tough encounters from being unavoidable, for the thrower a range limit plus reasonable movement speed should be sufficient.
That having been said, this should also be pretty deep in the evolution chain.

**Comment by RogueYun** [comment]
> That having been said, this should also be pretty deep in the evolution chain.

I thought about a bone throwing skeleton once, one that would either pickup and throw nearby bones or damage themselves to throw their own bones. It might be something more simple that can (eventually) become something more terrifying.

**Comment by papersplease** [comment]
Honestly, I _love_ the idea, but without all the complications. Just make it so that they try to get within the melee range as usual, but either crush or throw anything that gets in their way to stomp you - zombies, furniture or the two-by-fours left of it, etc. Not necessarily at the player, the direction can be random. The player can avoid all this by controlling the hulk's line of sight and ensuring that nothing gets in his way.

**Comment by aeoo** [comment]
Honestly I never thought my idea was perfect in every way or anything.  I don't necessarily have any intelligent answers to anyone who saw problems in my idea.  If you think there is a problem, there probably is.

The *reason* behind what I said initially is basically to give zombies a ranged option against the player.

I was initially thinking these should not be like the other hulks, because they focus on throwing and so should be cowardly and relatively weak in melee.  Since they take a bit of time to charge up their throws, it should be possible but difficult to outmanoeuvre them.  Since they're not like the other hulks, I was thinking maybe slightly more rare than say the "tough zombie."  Maybe more common than a necromancer.

I emphasize it was only a thought, nothing more.  So I don't necessarily think the way I presented it is fully baked.  I just wanted to see if anyone wanted to think about it, nothing more.

Basically it's about a ranged option for zombies.  With that all my important cards are on the table. :)

**Comment by Night-Pryanik** [comment]
No discussion in over half a year, closing.
